lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1140,6 +1139,12 @@ WasmBytecodeGenerator::EmitMemAccess(WasmOp wasmOp, const WasmTypes::WasmType* s
     {
         throw WasmCompilationException(_u("Index expression must be of type I32"));
     }
+
+    if (offset >= m_module->GetMemory()->maxSize)
+    {
+        throw WasmCompilationException(_u("Index is out of bounds"));
```

Comment:
We are generating this bytecode sequence : 
    017d   Ld_IntConst        I3  int:-1
    0183   Add_Int              I2  I2  I3 
    0187   LdArrWasm        I2 = HEAP32[I2]

When index + offset overflow, by the time we execute LdArrWasm, the value would be wrapped around. And we cant trap. We might need a different sequence here.

Adding the above check allowed me to catch such cases during bytecode generation. 
Should we generate a different sequence instead ?